### PR TITLE
Refactor LayerUI

### PR DESCRIPTION
**Pull request by ad1992**
* Pass named function to react.memo so that in dev tools the name shows up instead of showing anonymous 
* Make LayerUI default export
* Break the functional component into multiple render methods

### CODE REVIEW

File: src/components/LayerUI.tsx [hunk 1]
```diff
@@ -40,210 +40,236 @@ interface LayerUIProps {
...
-            onClose={() => setAppState({ errorMessage: null })}
-          />
-        )}
-        {appState.showShortcutsDialog && (
-          <ShortcutsDialog
-            onClose={() => setAppState({ showShortcutsDialog: null })}
+    );
+  }
+
+  function renderCanvasActions() {
```
Comment (lipis) [inline src/components/LayerUI.tsx:96]: could you use arrow functions?

File: src/components/LayerUI.tsx [hunk 2]
```diff
@@ -40,210 +40,236 @@ interface LayerUIProps {
...
-          />
-        )}
-        {appState.showShortcutsDialog && (
-          <ShortcutsDialog
-            onClose={() => setAppState({ showShortcutsDialog: null })}
+    );
+  };
+
+  const renderCanvasActions = () => {
+    return (
```
Comment (lipis) [inline src/components/LayerUI.tsx:97]: this could be simplified since it's only return

File: src/components/LayerUI.tsx [hunk 3]
```diff
@@ -40,210 +40,236 @@ interface LayerUIProps {
...
+            </Stack.Row>
+            {actionManager.renderAction("changeViewBackgroundColor")}
+          </Stack.Col>
+        </Island>
+      </Section>
+    );
+  };
+
+  const renderSelectedShapeActions = () => {
+    return (
```
Comment (lipis) [inline src/components/LayerUI.tsx:123]: return not needed since it's the only thing

File: src/components/LayerUI.tsx [hunk 4]
```diff
@@ -40,210 +40,236 @@ interface LayerUIProps {
...
-      return ret;
-    };
-    const prevAppState = getNecessaryObj(prev.appState);
-    const nextAppState = getNecessaryObj(next.appState);
-
-    const keys = Object.keys(prevAppState) as (keyof Partial<AppState>)[];
+  };
 
+  const renderFooter = () => {
     return (
```
Comment (lipis) [inline src/components/LayerUI.tsx:191]: same :)

**Comment by ad1992** [comment]
@lipis the lint for pull request is failing but the changes are due to running prettier the file App.tsx.

**Comment by lipis** [comment]
run `npm install` we have a new version of Prettier

**Comment by ad1992** [comment]
@lipis @vjeux @dwelle did you get a chance to check this ?

**Comment by dwelle** [comment]
@ad1992 will do at my earliest convenience. Do you need help resolving the conflicts?

**Comment by ad1992** [comment]
@dwelle I have resolved the conflicts. Please let me know if it's fine. It would be of great help if you could reply in https://github.com/excalidraw/excalidraw/issues/1196, I am working on creating npm package for excalidraw as I am integrating into a web app and need to customize the UI as well. That's why I wanted to know what you guys think so I know whether I should open a PR for the same. 
Thank you.

**Comment by dwelle** [comment]
Ok, looking good. Thanks.

As for #1196, will check tomorrow, but that one is pretty involved.

**Comment by ad1992** [comment]
Thanks @dwelle .Looking forward to your reply. I wanted to avoid creating the package in the forked repo so I just wanted the confirmation If I can open the PR here.

**Pull request merged**